go/host/host.go

```diff
@@ -522,9 +522,9 @@ func (h *host) publishRollup(producedRollup common.ExtRollup) {
 
 func (h *host) storeBlockProcessingResult(result *common.BlockSubmissionResponse) {
 	// only update the host rollup headers if the enclave has found a new rollup head
-	if result.FoundNewHead {
+	if result.NewRollupHeader != nil {
 		// adding a header will update the head if it has a higher height
-		headerWithHashes := common.HeaderWithTxHashes{Header: result.RollupHead, TxHashes: result.ProducedRollup.TxHashes}
+		headerWithHashes := common.HeaderWithTxHashes{Header: result.NewRollupHeader, TxHashes: result.HeadRollup.TxHashes}
```

Comment:
I think this is a (pre-existing but fairly recent) bug. ProducedRollup is the rollup *after* NewRollupHeader, I just put a log message in and verified that result.RollupHead was a different height to result.ProducedRollup

There's an important distinction between the produced rollup and the newly ingested rollup (the former is a candidate rollup that's just been produced to publish, the latter is a rollup that was just discovered in an L1 block). I don't think we can removed the field from the Response object.